src/main/java/address/browser/BrowserManager.java

```diff
@@ -58,32 +61,36 @@
     };
 
     public BrowserManager(ObservableList<ReadOnlyViewablePerson> filteredPersons) {
-        initializeBrowser();
         this.selectedPersonUsername = new SimpleStringProperty();
         this.filteredPersons = filteredPersons;
     }
 
     /**
-     * Initialize the application to use jxBrowser.
+     * Initialize the browser managed by the browser manager.
      * This must be called in a non-ui thread.
      */
-    public static void initializeJxBrowserEnvironment(){
-        if (Environment.isMac()) {
-            BrowserCore.initialize();
+    public static void initBrowser(){
+        if (type == BrowserType.FULL_FEATURE_BROWSER) {
+            if (Environment.isMac()) {
+                BrowserCore.initialize();
+            }
+            logger.debug("Suppressing browser logs");
+            LoggerProvider.setLevel(Level.SEVERE);
         }
```

Comment:
This is better, but even better if you use a polymorphic call.
i.e. init() method is implemented in both types of browsers and you call the init() method here without knowing which type of a browser it is.